docs/api/specs/browserResidentDesktopAgents.md

```diff
@@ -0,0 +1,240 @@
+
+Finally, Desktop Agents SHOULD retain instance details for applications that have closed as they may appear to close during navigation events, or may navigate away and then navigate back. By retaining the instance data (`instanceId`, `instanceUuid` and `WindowProxy`) the same instance identity can be maintained or reissued. There is no standard length of time that such details should be retained, hance, Desktop Agents MAY determine for themselves how long to retain instance details for closed instances.
+
+## Responding to app communications with Desktop Agent Communication Protocol (DACP)
+
+After validating an application's identity and any instance identity to be reused, the Desktop Agent is ready to support communication with the application. This is achieved via the [Desktop Agent Communication Protocol](./desktopAgentCommunicationProtocol) (DACP) over the `MessageChannel` established in the previous steps.
+
+## Implementing DAs in hidden iframes
+
+As described above, DA providers can leverage hidden iframes to establish a communication mechanism that is independent of a Parent window. This approach allows apps to connect to a DA even when they were not opened by that DA.
+
+The hidden iframe url can be provided in two ways:
+
+1) By a Parent window - This allows DAs to redirect communications to via a hidden iframe that loads a known URL. The main benefit of this approach is that it can allow a system to continue to operate even if the parent window is closed.
+
+2) By a `failover` function - When no parent DA can be found (such as when a tab is opened directly by an end user) then a failover function can create a hidden iframe and return a reference to it (a `WindowProxy`) to initiate communication with via the WCP and DACP in the same way as we do with a parent window. Alternatively, a `DesktopAgent` implementation maybe loaded directly and returned from the failover function, which `getAgent()` will pass-through.
+
+## Channel Selector and Intent Resolver User Interfaces
+
+Channel Selector and Intent Resolver user-interfaces are normally provided by Desktop Agents to applications. However, when running in a web browser a DA may not have the ability to present a channel selector in a window that has been opened with `window.open()` and it may be challenging to display a secondary window over the application when needed (due to pop-up blocking and user preferences).
+
+The `getAgent()` implementation can facilitate the injection and management of iframes in an application window. An app may use the optional `channelSelector` and `intentResolver` parameters to the `getAgent()` to indicate whether or not they need these interfaces (for example they may not raise intents or may provide their own resolver UI based on the DA API's `findIntent` or `findIntentsForContext` functions so they don't need a DA-provided interface). These parameters are forwarded onto the Desktop Agent in the `WCP1Hello` connection message.
```

Comment:
```suggestion
The `getAgent()` implementation can facilitate the injection and management of iframes in an application window. An app may provide the optional `channelSelector` and `intentResolver` parameters to the `getAgent()` to indicate whether or not they need these interfaces. For example, the apps may not raise intents. Some apps may also resolve intents internally by leveraging the DA' `findIntent` or `findIntentsForContext` API methods. In these situations, the apps won't need a DA-provided interface. Once an app calls `getAgent()`, the parameters the app provides are forwarded onto the Desktop Agent in the `WCP1Hello` connection message.
```

Reply:
Thanks, tweaked your suggestion and applied it.